### PR TITLE
Ecal Phase 2 TP algo EcalFenix class renames

### PR DESCRIPTION
**Pull request by thomreis**
#### PR description:

This PR renames the EcalFenix classes used in the Phase 2 ECAL TP implementation to distinguish them from the Phase 1 TP classes. The classes are renamed to EcalEBFenix* to follow the package name they are in.

While running the matrix tests for PR #33349 the Phase 2 WF showed differences to the baseline. It was noticed that the Phase 1 EcalFenix headers in `SimCalorimetry/EcalTrigPrimAlgos` and the Phase 2 EcalFenix headers in `SimCalorimetry/EcalEBTrigPrimAlgos` were previously identical, including the include guards.
With the changes to the Phase 1 TP emulator in PR #33349 the Phase 2 functions were used in some WFs in the Phase 1 code instead of the Phase 1 ones, leading to failing tests for Phase 2 WFs.

#### PR validation:

Passes Phase 2 WF 23234.0 matrix test.

### CODE REVIEW

**Comment by thomreis** [comment]
type bugfix

**Comment by srimanob** [comment]
Please test

**Comment by cecilecaillol** [comment]
+l1

File: SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBFenixLinearizer.h [hunk 1]
```diff
@@ -59,7 +59,7 @@ void EcalFenixLinearizer::process(const T &df, std::vector<int> &output_percry)
   // 1  8  11 18 21
   // 0  9  10 19 20
 
-  //std::cout << " EcalFenixLinearizer::process(const  .. DataFrame " << std::endl;
+  //std::cout << " EcalEBFenixLinearizer::process(const  .. DataFrame " << std::endl;
   for (int i = 0; i < df.size(); i++) {
```
Comment (srimanob) [inline SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBFenixLinearizer.h:63]: Why do you need to keep this loop on, it seems it does nothing here.

File: SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBFenixLinearizer.h [hunk 2]
```diff
@@ -69,7 +69,7 @@ void EcalFenixLinearizer::process(const T &df, std::vector<int> &output_percry)
     output_percry[i] = process();
   }
 
-  //std::cout << " EcalFenixLinearizer::process(const  .. Final output " << std::endl;
+  //std::cout << " EcalEBFenixLinearizer::process(const  .. Final output " << std::endl;
   for (int i = 0; i < df.size(); i++) {
```
Comment (srimanob) [inline SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBFenixLinearizer.h:73]: Same comment as above.

File: SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBFenixLinearizer.h (hunk 1 above)
Comment (thomreis) [reply SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBFenixLinearizer.h:63]: Tbh I did not look at the code functionality itself. I just made the minimal changes to fix the issue. I can remove the loop if needed.

Comment (thomreis) [reply SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBFenixLinearizer.h]: Hi @srimanob I have pushed a new commit that removes the two useless loops. Could you stop the ongoing tests and start new ones please?

Comment (srimanob) [reply SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBFenixLinearizer.h]: Thanks.

Comment (thomreis) [reply SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBFenixLinearizer.h]: Thanks. Could you also launch the tests for #33349 in combination with this PR if you are on it?

**Comment by srimanob** [comment]
Please test

**Comment by srimanob** [comment]
SimCalorimetry/EcalEBTrigPrimAlgos/src/EcalEBTrigPrimTestAlgo.cc is picked by the clang tidy checks for deprecated calls, https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c7b2e2/14169/llvm-analysis/cmsclangtidy.txt

Could this be updated also, @thomreis? Thanks.

**Comment by thomreis** [comment]
Sure. I will implement the necessary changes.

**Comment by thomreis** [comment]
Hi @srimanob the deprecated calls should have been removed now. I followed what has been done in #33062 for the Phase 1 code.

**Comment by srimanob** [comment]
urgent

**Comment by srimanob** [comment]
Please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by thomreis** [comment]
Tests look OK as before. The 140.53 WF differences seem DAS related. If you are happy with the PR can you sign please @srimanob ?

**Comment by srimanob** [comment]
+Upgrade

Test together with https://github.com/cms-sw/cmssw/pull/33349 seems to give the expected result.

**Comment by silviodonato** [comment]
+1

**Pull request merged**